src/libraries/System.Reflection.Emit/tests/PersistableAssemblyBuilder/AssemblySaveCustomAttributeTests.cs

```diff
@@ -0,0 +1,358 @@
+                TypeBuilder tb = mb.DefineType(type.FullName, type.Attributes, type.BaseType);
+                typeAttributes.ForEach(tb.SetCustomAttribute);
+
+                DefineMethodsAndSetAttributes(methodAttributes, tb, type.IsInterface ? type.GetMethods() : type.GetMethods(BindingFlags.DeclaredOnly));
+                DefineFieldsAndSetAttributes(fieldAttributes, type.GetFields(), tb);
+            }
+        }
+
+        private static void DefineFieldsAndSetAttributes(List<CustomAttributeBuilder>? fieldAttributes, FieldInfo[] fields, TypeBuilder tb)
+        {
+            foreach (FieldInfo field in fields)
+            {
+                FieldBuilder fb = tb.DefineField(field.Name, field.FieldType, field.Attributes);
+                fieldAttributes.ForEach(fb.SetCustomAttribute);
+            }
+        }
+
+        private static void DefineMethodsAndSetAttributes(List<CustomAttributeBuilder> methodAttributes, TypeBuilder tb, MethodInfo[] methods)
+        {
+            foreach (var method in methods)
+            {
+                MethodBuilder meb = tb.DefineMethod(method.Name, method.Attributes, method.CallingConvention, method.ReturnType, null);
+                methodAttributes.ForEach(meb.SetCustomAttribute);
+            }
+        }
+
+        [Fact]
+        public void CreateStructWithPseudoCustomAttributesTest()
+        {
+            using (TempFile file = TempFile.Create())
+            {
+                Type type = typeof(StructWithFields);
+                List<CustomAttributeBuilder> typeAttributes = new() { new CustomAttributeBuilder(typeof(SerializableAttribute).GetConstructor(Type.EmptyTypes), new object[] { }),
+                                                              new CustomAttributeBuilder(typeof(StructLayoutAttribute).GetConstructor(new Type[] { typeof(LayoutKind) }), new object[] { LayoutKind.Explicit },
+                                                                    typeof(StructLayoutAttribute).GetFields() , new object[]{32, 64, CharSet.Unicode}),
+                                                              new CustomAttributeBuilder(s_guidPair.con, s_guidPair.args),
+                                                              new CustomAttributeBuilder(typeof(SpecialNameAttribute).GetConstructor(Type.EmptyTypes), new object[] { })
+                                                            };
+                CustomAttributeBuilder[] fieldAttributes = new[] { new CustomAttributeBuilder(typeof(NonSerializedAttribute).GetConstructor(Type.EmptyTypes), new object[] { }),
+                                                              new CustomAttributeBuilder(typeof(FieldOffsetAttribute).GetConstructor(new Type[] { typeof(int) }), new object[] { 2 }),
+                                                              new CustomAttributeBuilder(s_guidPair.con, s_guidPair.args),
+                                                              // TODO: Need to support the UnmanagedType type
```

Comment:
What is needed to support the UnmanagedType type?

Reply:
Before non-primitive parameter type was not supported, but now [with](https://github.com/dotnet/runtime/pull/85112) it should work, enabling this test scenario

Reply:
Because it was not tested the `MarshalAs` pseudo attribute was left out of proper handling 😝. It has a lot of fields and scenarios depending on `UnmanagedType` 😓, but now its done, please review 😄.